### PR TITLE
cmd: add support for `ENV` command

### PR DESCRIPTION
**Pull request by Delta456**
Implements https://github.com/charmbracelet/vhs/issues/419

image

```haskell
Output examples/demo.gif
Env PROMPT "〉"

Require echo

Set Shell "bash"
Set FontSize 32
Set Width 1200
Set Height 600

Type "echo 'Welcome to VHS!'" Sleep 500ms  Enter

Sleep 5s
```

and also allows adding environment variables.

image

```haskell
Env TESTING "this works"

Type "echo $TESTING"
Enter
Sleep 3s
```

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -248,6 +249,20 @@ func ExecuteCopy(c parser.Command, _ *VHS) {
...
+	_ = os.Setenv(c.Options, c.Args)
+
+	if c.Options == "PROMPT" {
+		if v.Options.Shell.Env != nil {
+			v.Options.Shell.Env[0] = strings.ReplaceAll(v.Options.Shell.Env[0], ">", c.Args)
+		} else if v.Options.Shell.Command != nil {
+			last := len(v.Options.Shell.Command) - 1
+			v.Options.Shell.Command[last] = strings.ReplaceAll(v.Options.Shell.Command[last], ">", c.Args)
+		}
+	}
```
Comment (Delta456) [inline command.go:263]: I couldn't find a better way to set up during the evaluation time. This also feels very hacky. I would love to know a better way to do this.

File: command.go [hunk 2]
```diff
@@ -248,6 +249,20 @@ func ExecuteCopy(c parser.Command, _ *VHS) {
 	_ = clipboard.WriteAll(c.Args)
 }
 
+// ExecuteEnv sets env with given key-value pair.
+func ExecuteEnv(c parser.Command, v *VHS) {
+	_ = os.Setenv(c.Options, c.Args)
```
Comment (Delta456) [inline command.go:254]: Need to also un-set these environment variables after they are used(?)

Comment (maaslalani) [reply command.go:254]: Does this affect the actual users session? I imagine it doesn't and will disappear after the recording is over. If it does persist after the session is done then we could first save the user environment and then set the environment to the `oldEnvironment` afterwards.

File: command.go (hunk 1 above)
Comment (Delta456) [reply command.go:263]:
```go
package main

import (
	"fmt"
	"os"
)

// Supported shells of VHS
const (
	bash       = "bash"
	cmdexe     = "cmd"
	fish       = "fish"
	nushell    = "nu"
	powershell = "powershell"
	pwsh       = "pwsh"
	zsh        = "zsh"
)

// Shell is a type that contains a prompt and the command to set up the shell.
type Shell struct {
	Command []string
	Env     []string
}

var prompt = func() string {
	prompt := os.Getenv("PROMPT")
	if prompt == "" {
		return "> "
	}
	return prompt
}()

// Shells contains a mapping from shell names to their Shell struct.
var Shells = map[string]Shell{
	bash: {
		Env:     []string{fmt.Sprintf("PS1=\\[\\e[38;2;90;86;224m\\]%s \\[\\e[0m\\]", prompt), "BASH_SILENCE_DEPRECATION_WARNING=1"},
		Command: []string{"bash", "--noprofile", "--norc", "--login", "+o", "history"},
	},
	zsh: {
		Env:     []string{fmt.Sprintf(`PROMPT=%%F{#5B56E0}%s %%F{reset_color}`, prompt)},
		Command: []string{"zsh", "--histnostore", "--no-rcs"},
	},
	fish: {
		Command: []string{
			"fish",
			"--login",
			"--no-config",
			"--private",
			"-C", "function fish_greeting; end",
			"-C", fmt.Sprintf(`function fish_prompt; set_color 5B56E0; echo -n "%s "; set_color normal; end`, prompt),
		},
	},
	powershell: {
		Command: []string{
			"powershell",
			"-NoLogo",
			"-NoExit",
			"-NoProfile",
			"-Command",
			fmt.Sprintf(`Set-PSReadLineOption -HistorySaveStyle SaveNothing; function prompt { Write-Host '%s' -NoNewLine -ForegroundColor Blue; return ' ' }`, prompt),
		},
	},
	pwsh: {
		Command: []string{
			"pwsh",
			"-Login",
			"-NoLogo",
			"-NoExit",
			"-NoProfile",
			"-Command",
			fmt.Sprintf(`Set-PSReadLineOption -HistorySaveStyle SaveNothing; Function prompt { Write-Host -ForegroundColor Blue -NoNewLine '%s'; return ' ' }`, prompt),
		},
	},
	cmdexe: {
		Command: []string{"cmd.exe", "/k", fmt.Sprintf("prompt=^%s ", prompt)},
	},
	nushell: {
		Command: []string{"nu", "--execute", "$env.PROMPT_COMMAND = {''}"},
	},
}
```

Was trying something like this but it didn't work.

File: command.go (hunk 2 above)
Comment (Delta456) [reply command.go:254]: Ok it doesn't affect and disappears.

**Comment by Delta456** [comment]
Ready for review

File: command.go (hunk 1 above)
Comment (maaslalani) [inline command.go:263]: Let's remove this hack for now and keep thinking how to solve `PROMPT`s

Comment (maaslalani) [inline command.go:263]:
```suggestion
```

Comment (Delta456) [reply command.go]: Sure

File: README.md [hunk 3]
```diff
@@ -738,6 +739,35 @@ Sleep 500ms
...
+
+Set Shell "bash"
+Set FontSize 32
+Set Width 1200
+Set Height 600
+
+Type "echo 'Custom prompt'" Sleep 500ms  Enter
+
+Sleep 5s
+```
```
Comment (maaslalani) [inline README.md:770]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -738,6 +739,18 @@ Sleep 500ms
 Paste
 ```
 
+### Env
+
+`Env` command sets the environment variable via key-value pair.
+
+```elixir
+Env KEY "the command is working"
```
Comment (maaslalani) [inline README.md:747]:
```suggestion
Env HELLO WORLD
```

File: README.md [hunk 5]
```diff
@@ -738,6 +739,18 @@ Sleep 500ms
...
+
+Type "echo $KEY"
```
Comment (maaslalani) [inline README.md:749]:
```suggestion
Type "echo $HELLO"
```

**Comment by maaslalani** [comment]
Awesome work @Delta456, thank you so much!

File: command.go [hunk 6]
```diff
@@ -248,6 +249,12 @@ func ExecuteCopy(c parser.Command, _ *VHS) {
 	_ = clipboard.WriteAll(c.Args)
 }
 
+// ExecuteEnv sets env with given key-value pair.
+func ExecuteEnv(c parser.Command, v *VHS) {
```
Comment (maaslalani) [inline command.go:253]:
```suggestion
func ExecuteEnv(c parser.Command, _ *VHS) {
```

File: command.go [hunk 7]
```diff
@@ -248,6 +249,12 @@ func ExecuteCopy(c parser.Command, _ *VHS) {
 	_ = clipboard.WriteAll(c.Args)
 }
 
+// ExecuteEnv sets env with given key-value pair.
+func ExecuteEnv(c parser.Command, _ *VHS) {
+	_ = os.Setenv(c.Options, c.Args)
+
```
Comment (maaslalani) [inline command.go:255]:
```suggestion
```

**Comment by rahji** [comment]
Can we get a new Release so this feature is included when someone does `go install github.com/charmbracelet/vhs@latest`? PS the helix editor's syntax highlighting file will need updating.

**Comment by SVendittelli** [comment]
I second the request for a new release. I install using `winget` and I'd like to be able to use the new `Env` functionality.

**Pull request merged**